testplan/common/utils/table.py

```diff
@@ -1,40 +1,114 @@
 """Utilities for working with tables."""
 import six
-import collections
-
-
-def all_are(objs, *are_what):
-    # type: (Iterable[Any], Type) -> bool
-    return all(isinstance(obj, are_what) for obj in objs)
+from six.moves import range
+from collections import OrderedDict
+
+# pylint: disable=import-error, no-name-in-module
+if six.PY3:
+    from collections.abc import Container, Mapping, Sequence
+else:
+    from collections import Container, Mapping, Sequence
+# pylint: enable=import-error, no-name-in-module
+
+_TP_BLANK_CELL = "_TP_BLANK_CELL"
+
+
+def _is_nonstr_sequence(obj):
+    return isinstance(obj, Sequence) and not isinstance(obj, six.string_types)
+
+
+def _table_from_list_of_lists(list_of_lists):
+    # type: (Sequence[Sequence[Any]]) -> List[List[Any]]
+    std_tbl = []
+    # all that needs to happen here is that each nested list must be padded to
+    # the length of the longest list
```

Comment:
I would suggest we be more strict on user input. If user pass in a list of list, all sub-lists shall be of uniform length. It's a guess to pad at the end anyway - so better not do that at all.